From 52147e71606abac26a0358285573b914427ef94b Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Fri, 4 Mar 2011 14:50:28 +1000
Patch-mainline: 2.6.39
References: fate#310916
Git-commit: e73f88af66fcc50083fae4b7e1c39b469179a97a
Subject: [PATCH 2195/2588] drm: add cap bit to denote if dumb ioctl is
 available or not.

This allows libkms to make an easier decision.

Signed-off-by: Dave Airlie <airlied@redhat.com>
(cherry picked from commit e73f88af66fcc50083fae4b7e1c39b469179a97a)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/drm_ioctl.c |    8 ++++++++
 include/drm/drm.h           |    2 ++
 2 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/drivers/gpu/drm/drm_ioctl.c b/drivers/gpu/drm/drm_ioctl.c
index d6de9d0..7f6912a 100644
--- a/drivers/gpu/drm/drm_ioctl.c
+++ b/drivers/gpu/drm/drm_ioctl.c
@@ -275,6 +275,14 @@ int drm_getcap(struct drm_device *dev, void *data, struct drm_file *file_priv)
 	struct drm_get_cap *req = data;
 
 	req->value = 0;
+	switch (req->capability) {
+	case DRM_CAP_DUMB_BUFFER:
+		if (dev->driver->dumb_create)
+			req->value = 1;
+		break;
+	default:
+		return -EINVAL;
+	}
 	return 0;
 }
 
diff --git a/include/drm/drm.h b/include/drm/drm.h
index da4efd1..9ac4313 100644
--- a/include/drm/drm.h
+++ b/include/drm/drm.h
@@ -752,6 +752,8 @@ struct drm_event_vblank {
 	__u32 reserved;
 };
 
+#define DRM_CAP_DUMB_BUFFER 0x1
+
 /* typedef area */
 #ifndef __KERNEL__
 typedef struct drm_clip_rect drm_clip_rect_t;
-- 
1.7.6

